Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -706,11 +712,14 @@ - (void)addAttachmentUploadViewIfNecessary:(UIView *)attachmentView
 {
     OWSAssert(attachmentView);
     OWSAssert(attachmentStateCallback);
+    OWSAssert(self.attachmentStream);
 
     if (self.isOutgoing) {
-        self.attachmentUploadView = [[AttachmentUploadView alloc] initWithAttachment:self.attachmentStream
-                                                                           superview:attachmentView
-                                                             attachmentStateCallback:attachmentStateCallback];
+        if (!self.attachmentStream.isUploaded) {
```

Comment:
Small optimization - don't both with attachment upload view if we don't need it.

Reply:
Will this show for failed uploads which aren't "in progress"?

Reply:
Nope, there's logic in the upload view to hide in that case.